lib/segment/src/payload_storage/query_checker.rs

```diff
@@ -234,6 +240,8 @@ mod tests {
             "rating": vec![3, 7, 9, 9],
             "color": "red",
             "has_delivery": true,
+            "parts": [],
+            "packaging": null
```

Comment:
Is it possible to add a test for a field that is `[null]`?

Reply:
How is the filtering supposed to behave in that case?

Reply:
I guess `[null]` is equivalent to an empty array, so not null.
WDYT?

Reply:
With the current implementation, it will not be matched with neither `is_empty` nor `is_null`.
If you want it to be matched with `is_empty` that's fine. However, what about `[null, [null]]`, is that even a valid payload? if so, should it be equivalent to empty as well?

Reply:
Forget about my comment, I don't think it is necessary to test this.
Sorry for derailing your research.

Reply:
It's alright, I added it anyway to test for any future changes regarding this case.